### PR TITLE
Upgrade to nixpkgs 19.03 (again)

### PR DESCRIPTION
**Pull request by cocreature**
I reverted the first attempt at this since we saw flaky cache downloads on Linux CI builds. However, these didn’t seem to be caused by the update (the build directly after merging passed on Linux and we saw the same failure even after reverting) and I added retries. So I’ll try this again.

Marking as draft until I’m sure that it doesn’t break things.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by cocreature** [comment]
I did actually manage to trigger the error (it makes sense that it was triggered by this PR even if it is not the cause as a nixpkgs upgrade will cause a lot of cache downloads and those are the cause) and it turned out my restarting logic was broken. After fixing it, I’ve now done 5 rebuilds and all of them succeeded (past the dev-env step) so I’m pretty confident that this won’t mess things up.

File: ledger/api-server-damlonx/src/main/scala/com/daml/ledger/api/server/damlonx/services/DamlOnXCommandCompletionService.scala [hunk 1]
```diff
@@ -126,8 +126,9 @@ object DamlOnXCommandCompletionService {
   def create(indexService: IndexService)(
       implicit ec: ExecutionContext,
       mat: Materializer,
-      esf: ExecutionSequencerFactory)
```
Comment (cocreature) [inline ledger/api-server-damlonx/src/main/scala/com/daml/ledger/api/server/damlonx/services/DamlOnXCommandCompletionService.scala]: These changes are caused by a `scalafmt` upgrade.

File: ci/dev-env-install.sh [hunk 2]
```diff
@@ -39,7 +39,7 @@ step "Building dev-env dependencies"
 # to make it retry itself. See https://github.com/NixOS/nix/issues/2794 for the issue requesting
 # this feature.
 for i in `seq 10`; do
-    nix-build nix -A tools -A cached 2>&1 | tee nix_log
+    nix-build nix -A tools -A cached 2>&1 | tee nix_log || true
```
Comment (neil-da) [inline ci/dev-env-install.sh]: If Nix does fail, do we get a failing exit code? Or just continue always now?

Comment (cocreature) [reply ci/dev-env-install.sh]: We don’t look at the exit code but we look at the error message so we don’t always retry.

Comment (aherrmann-da) [reply ci/dev-env-install.sh]: @cocreature How about something like `if !nix-build ...` or `unset failed; nix-build ... || failed=1` to make sure that we don't silently ignore `nix-build` errors which might cause confusing errors down the line? You can then still check for the log content to decide whether to retry.

File: nix/bazel.nix [hunk 3]
```diff
@@ -23,6 +22,13 @@ rec {
     zip
     ;
 
+  # We need to have a file in GOPATH that we can use as
+  # root_file in go_wrap_sdk.
+  go = pkgs.go.overrideAttrs (oldAttrs: {
+    doCheck = false;
+    postFixup = ''touch $out/share/go/ROOT'';
```
Comment (aherrmann-da) [inline nix/bazel.nix]: Did they drop the `README.md` or why is this necessary?

File: nix/tools/bazel-deps/default.nix [hunk 4]
```diff
@@ -74,7 +74,7 @@ buildBazelPackage rec {
       find . -type d -empty -delete
     '';
 
-    sha256 = "0xnwg847zkxf6vfmhsi1g16a4q78rgaa87prwcqhxl9psyl9r30c";
+    sha256 = "0m36zfrbwxzjm5vpvrjgs3vmgk90x010wiki5aspjxs0n8l9dxvh";
```
Comment (aherrmann-da) [inline nix/tools/bazel-deps/default.nix]: Did you check if the `fetchAttrs` hash of `bazel-watcher` would change as well?

File: nix/nixpkgs.nix [hunk 5]
```diff
@@ -43,48 +43,6 @@ let
           wrapProgram $out/bin/pg_tmp --prefix PATH : ${pkgs.postgresql}/bin:$out/bin
         '';
       });
-      buildBazelPackage = pkgs.callPackage ./overrides/buildBazelPackage {
```
Comment (aherrmann-da) [inline nix/nixpkgs.nix]: Why is it safe to remove this override on MacOS? Is it the default in 19.03 now?

File: ci/dev-env-install.sh (hunk 2 above)
Comment (cocreature) [reply ci/dev-env-install.sh]: Good idea, changed.

File: nix/bazel.nix (hunk 3 above)
Comment (cocreature) [reply nix/bazel.nix]: Yep, the README.md is gone.

File: nix/tools/bazel-deps/default.nix (hunk 4 above)
Comment (cocreature) [reply nix/tools/bazel-deps/default.nix]: I hadn’t but now I did and it stayed the same.

File: nix/nixpkgs.nix (hunk 5 above)
Comment (cocreature) [reply nix/nixpkgs.nix]: Yep, afaict all our patches have since been fixed upstream.

File: ci/dev-env-install.sh [hunk 6]
```diff
@@ -38,13 +38,18 @@ step "Building dev-env dependencies"
 # repeat this a few times. There does not seem to be an option that we can pass to nix
 # to make it retry itself. See https://github.com/NixOS/nix/issues/2794 for the issue requesting
 # this feature.
+NIX_FAILED=0
 for i in `seq 10`; do
```
Comment (neil-da) [inline ci/dev-env-install.sh]: I'm sceptical this complexity is worth it. I'd just retry 10 times, stopping on the first success or failing if we require all 10 times. Something as simple as `$A || $A || $A || $A || $A || $A || $A || $A` would make me happier - shell script should be feared.

Comment (cocreature) [reply ci/dev-env-install.sh]: If we retry 10 times and have e.g. an actual build failure in GHC, I have to wait for 10 hours until my build actually fails. As much as I dislike shell scripting, I still prefer it over that.

Comment (neil-da) [reply ci/dev-env-install.sh]: You're always in a hurry... Fair enough, that makes sense.

Comment (cocreature) [reply ci/dev-env-install.sh]: I prefer spending my time on more exiting things that build failures :wink:
